### PR TITLE
ThemeProvider: clean up public APIs and ensure default theme is always provided

### LINKED ISSUE #15212: React-Theme-Provider Storybook broken
### Environment Information

- **Package version(s)**: 0.14.0
- **Browser and OS versions**: (fill this out if relevant)

### Please provide a reproduction of the bug in a codepen:

git clone https://github.com/microsoft/fluentui.git
cd fluentui/packages/react-theme-provider
yarn
yarn run start

#### Actual behavior:

image

#### Expected behavior:

To not have property undefined errors on Storybook run

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Low

Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by xugao**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15212 
- [x] Include a change request file using `$ yarn change`

#### Description of changes
- Removed `FluentTheme` and `TeamsTheme` from `react-theme-provider`
- Ensure a full default theme is provided

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by ecraig12345 — APPROVED** [review]

File: packages/storybook/src/themes/v8/TeamsTheme.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { PartialTheme } from '../types';
+import { PartialTheme } from '@fluentui/react-theme-provider';
```
Comment (dzearing) [inline packages/storybook/src/themes/v8/TeamsTheme.ts:1]: intended?

File: packages/react-theme-provider/src/createTheme.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Theme, createTheme as legacyCreateTheme, PartialTheme } from '@fluentui/theme';
```
Comment (dzearing) [inline packages/react-theme-provider/src/createTheme.ts:1]: why is this called legacyCreateTheme here?

**Comment by khmakoto** [comment]
Should we remove the `getTokens` file in `react-theme-provider` as part of this PR?

**Review by khmakoto — APPROVED** [review]

**Comment by dzearing** [comment]
oof on the bundle size; is it because Button depends on ThemeProvider?

**Comment by xugao** [comment]
@dzearing - the bundle size is increased due to `createTheme` is added `useTheme` and `useTheme` is used by button

**Review by dzearing — APPROVED** [review]

**Pull request merged**